server/task.go

```diff
@@ -265,9 +265,11 @@ func (s *Server) patchTask(ctx context.Context, task *api.Task, taskPatch *api.T
 			}
 			oldStatement = payload.Statement
 			payload.Statement = *taskPatch.Statement
-			// We should update the schema version if we've updated the SQL, otherwise we will
-			// get migration history version conflict if the previous task has been attempted.
-			payload.SchemaVersion = common.DefaultMigrationVersion()
+			if payload.MigrationInfo == nil {
```

Comment:
What's the implication of `payload.MigrationInfo == nil`? Is that the issue hasn't been executed successfully?

Reply:
It means that the patch is from the UI workflow.

Reply:
added comments to explain this